sonic_platform_base/sonic_xcvr/api/public/sff8636.py

```diff
@@ -403,3 +404,46 @@ def get_lpmode(self):
         # Since typically optics come up by default set to high power, in this case,
         # power_override not being set, function will return high power mode.
         return power_set and power_override
+
+    def get_power_class(self):
+        '''
+        Retrieves power class of the module.
+
+        Returns:
+            int: Power class of the module, -1 if it fails
+        '''
+        power_class_str = self.xcvr_eeprom.read(consts.POWER_CLASS_FIELD)
+        power_class = -1
+
+        if power_class_str is None:
+            return power_class
+        if not power_class_str.startswith(self.POWER_CLASS_PREFIX):
+            return power_class
+
+        prefix_len = len(self.POWER_CLASS_PREFIX)
+        power_class = int(power_class_str[prefix_len:prefix_len + 1])
+
+        return power_class
+
+    def set_high_power_class(self, enable):
+        '''
+        This function sets high power class for the module if needed.
+        It is only applicable for power class >= 5.
+
+        Args:
+            enable (bool): True to enable high power class, False to disable
+
+        Returns:
+            bool: True if the provision succeeds, False if it fails
+        '''
+        power_class = self.get_power_class()
+        ret = True
+
+        if power_class < 5:
+            return ret
+        elif power_class >= 8:
+            ret = self.xcvr_eeprom.write(consts.HIGH_POWER_CLASS_ENABLE_CLASS_8, enable)
+        else:  # Power class 5, 6, 7
+            ret = self.xcvr_eeprom.write(consts.HIGH_POWER_CLASS_ENABLE_CLASS_5_TO_7, enable)
+
+        return ret
```

Comment:
@longhuan-cisco can we keep the set_high_power_class API as dumb. Let the SFF manager do the logic handling as to what power class to program?

Reply:
@longhuan-cisco the API `set_high_power_class()` is considered to be dumb, provides the eeprom write to enable the SFF-mgr requested power class. Its upto the SFF manager to read the advertisment to figure out what power class is supported by the module and only pass that as an argument to the API.